cmd/juju/backups/remove.go

```diff
@@ -58,18 +74,75 @@ func (c *removeCommand) Run(ctx *cmd.Context) error {
 			return err
 		}
 	}
-	client, err := c.NewAPIClient()
+
+	client, apiVersion, err := c.NewGetAPI()
 	if err != nil {
 		return errors.Trace(err)
 	}
 	defer client.Close()
 
-	err = client.Remove(c.ID)
-	if err != nil {
-		return errors.Trace(err)
+	if apiVersion < 2 {
+		if c.KeepLatest {
+			return errors.New("--keep-latest is not supported by this controller")
+		}
 	}
 
-	output := fmt.Sprintf("successfully removed: %v\n", c.ID)
-	ctx.Stdout.Write([]byte(output))
-	return nil
+	ids := []string{}
+	var keep string
+	if c.KeepLatest {
+		list, err := client.List()
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		ids, keep, err = parseList(list.List)
+		switch {
+		case err != nil:
+			return errors.Trace(err)
+		case len(ids) > 0:
+			break
+		case keep != "":
+			ctx.Warningf("no backups to remove, %v most current", keep)
+			return nil
+		default:
+			ctx.Warningf("no backups to remove")
+			return nil
+		}
+	} else {
+		ids = append(ids, c.ID)
+	}
+
+	results := client.Remove(ids)
+
+	for i, err := range results {
+		if err.Error != nil {
+			// Some errors do not provide enough info, let's try to fix that here.
+			err.Error.Message = fmt.Sprintf("failed to remove %v: %s", ids[i], err.Error.Message)
+			continue
+		}
+		ctx.Infof("successfully removed: %v\n", ids[i])
+	}
+	if c.KeepLatest {
+		ctx.Infof("kept: %v", keep)
+	}
+	return errors.Trace(params.ErrorResults{results}.Combine())
+}
+
+// parseList returns a list of IDs to be removed and the one ID to be kept.
+// Keep the latest ID based on Started.
+func parseList(list []params.BackupsMetadataResult) ([]string, string, error) {
+	if len(list) == 0 {
+		return nil, "", nil
+	}
+	latest := list[0]
+	retList := set.NewStrings()
+	for _, entry := range list[0:] {
```

Comment:
Why do we need `[0:]`, it doesn't look like you're modifying the slice, so `range list` should suffice?

Reply:
Latest shouldn't be added to the set, unless it's been proved to not be the latest.  The first latest is the top of the slice, so should be a 1 and not a 0.... hmmm